sdk/core/core-amqp/src/errors.ts

```diff
@@ -553,6 +553,18 @@ export function translate(err: AmqpError | Error): MessagingError {
 
   let error: MessagingError = err as MessagingError;
 
+  // InsufficientCreditError occurs when the number of credits available on Rhea link is insufficient.
+  // Since reasons for such shortage can be transient such as for pending delivery of messages, this is treated as a retryable error.
+  if (
+    // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
+    // Must do a name check until OperationTimeoutError is updated, and that doesn't break compatibility
```

Comment:
Nit: `OperationTimeoutError` should be `InsufficientCreditError`

Reply:
Updated